sched/semaphore/sem_tickwait.c

```diff
@@ -165,19 +145,24 @@ int nxsem_tickwait(FAR sem_t *sem, clock_t start, uint32_t delay)
  *
  ****************************************************************************/
 
-int nxsem_tickwait_uninterruptible(FAR sem_t *sem, clock_t start,
-                                   uint32_t delay)
+int nxsem_tickwait_uninterruptible(FAR sem_t *sem, uint32_t delay)
 {
-  int ret;
+  clock_t end = clock_systime_ticks() + delay;
 
-  do
+  for (; ; )
     {
       /* Take the semaphore (perhaps waiting) */
 
-      ret = nxsem_tickwait(sem, start, delay);
+      int ret = nxsem_tickwait(sem, delay);
+      if (ret != -EINTR)
+        {
+          return ret;
```

Comment:
let's add `break` and move `return ret;` to the end of the function scope

Reply:
Done.